### PR TITLE
Add operator  subcommand `upgrade` to istioctl

### PR DESCRIPTION
**Pull request by elfinhe**
Add operator (https://github.com/istio/operator) subcommand `upgrade` to istioctl.

Dependencies pulled by `GO111MODULE=on go get istio.io/operator@master` (https://github.com/istio/istio/wiki/Vendor-FAQ).

```
go get istio.io/operator@master ; go mod tidy ; go mod vendor
```

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[x] User Experience
[ ] Developer Infrastructure

Resolve: https://github.com/istio/istio/issues/17987

### CODE REVIEW

**Comment by howardjohn** [comment]
Yeah you should use the master branch in master, releas-1.4 branch in release-1.4

**Comment by elfinhe** [comment]
> Yeah you should use the master branch in master, releas-1.4 branch in release-1.4
which means I need to manually create 2 PRs for master and release 1.4?

Thanks John for the quick reply!

File: go.sum [hunk 1]
```diff
@@ -986,6 +1002,8 @@ istio.io/gogo-genproto v0.0.0-20190930162913-45029607206a h1:w7zILua2dnYo9CxImhp
 istio.io/gogo-genproto v0.0.0-20190930162913-45029607206a/go.mod h1:OzpAts7jljZceG4Vqi5/zXy/pOg1b209T3jb7Nv5wIs=
 istio.io/operator v0.0.0-20191016154438-acc189bf3318 h1:QyTRaCvR/FVBox6b1H9RB8aA60vFVBDblVMconK8hCY=
 istio.io/operator v0.0.0-20191016154438-acc189bf3318/go.mod h1:GoeYsVyEl7DcNBrVisUNqNe+yVpc3C+mRgUtenC+7Fw=
+istio.io/operator v0.0.0-20191017155015-6b3fd2596dbc h1:Dm15yLtGLFJxwGaR4spVLHQqheUPh4rBRPwN2sVRHQQ=
+istio.io/operator v0.0.0-20191017155015-6b3fd2596dbc/go.mod h1:3zAM53nNPyv9Z59m/61pCbZyTg9S5p+4dQCy9O8Ldi0=
```
[the remark this answers is not in the document]
Comment (elfinhe) [reply go.sum]: updated.

**Comment by elfinhe** [comment]
Seeing some fix on tests in master branch. 
Rebasing...

**Comment by elfinhe** [comment]
@howardjohn this is almost the same as https://github.com/istio/istio/pull/17990 (merged). But built with operator@master and istio@master.

**Pull request merged**